src/core.js

```diff
@@ -272,6 +272,16 @@ $.extend( $.validator, {
 				this.element( element );
 			}
 		},
+		oninput: function( element ) {
+			if ( element.name in this.submitted || element.name in this.invalid ) {
+				this.element( element );
+			} else {
+				return;
+			}
+		},
+
+		// DEPRECATED
+		// Please use `oninput` instead.
```

Comment:
I'm not the one who should take the decision, but I think it's ok to deprecate `onkeyup` and use `oninput` instead.
If this get deprecated, then we can remove it in `2.0.0` which should also define the supported browsers (`>IE9`).